### PR TITLE
https://github.com/intel/llvm/issues/6651: Fix lint

### PR DESCRIPTION
**Pull request by apstasen**
Fix lint by fetching enough commits (number of commits in pull request plus one) and using GITHUB_SHA that corresponds to sycl HEAD for pull_request_target instead of GITHUB_SHA^1 that corresponds to the first parent (previous state of sycl branch) of merge commit for pull_request used before.

### CODE REVIEW

**Review by aelovikov-intel — APPROVED** [review]
All owners are unavailable, so my LGTM is the most we can get.

The Lint task in this PR pre-commit shows the issue is fixed with this PR applied.

@againull  can you please merge this in using your maintainer rights?

File: .github/workflows/sycl_precommit.yml
Comment (bader) [inline .github/workflows/sycl_precommit.yml:36]: This approach pulls a lot of unnecessary commits. We need to pull just 2 commits: head of tested branch and merge-base with sycl branch. Before your changes we used pull request merge commit, so using fetch-depth: 2 was reasonable way to pull these two required commits.
Now as we don't use merge commit, we should probably use fetch-depth: 1, but fetch only required refs.

We have pulldowns from the community, which add a lot of commits to the sycl branch.

Comment (pvchupin) [reply .github/workflows/sycl_precommit.yml:36]: Tagging @apstasen

Comment (apstasen) [reply .github/workflows/sycl_precommit.yml:36]: We only fetch commits that belong PR branch (from the point when it was branched from sycl branch). If we use fetch depth 1, then git tree used in git clang format will not work properly since sycl HEAD and PR HEAD will not have common history. PR_FETCH_DEPTH is minimum possible number of commits we need.

**Pull request merged**
